command/volume_status.go

```diff
@@ -147,17 +148,61 @@ func (c *VolumeStatusCommand) Run(args []string) int {
 		id = args[0]
 	}
 
+	opts := formatOpts{
+		verbose:  c.verbose,
+		short:    c.short,
+		length:   c.length,
+		json:     c.json,
+		template: c.template,
+	}
+
 	switch typeArg {
-	case "csi", "":
+	case "csi":
 		if nodeID != "" || nodePool != "" {
 			c.Ui.Error("-node and -node-pool can only be used with -type host")
 			return 1
 		}
-		return c.csiStatus(client, id)
+		if err := c.csiVolumeStatus(client, id, opts); err != nil {
+			c.Ui.Error(err.Error())
+			return 1
+		}
 	case "host":
-		return c.hostVolumeStatus(client, id, nodeID, nodePool)
+		if err := c.hostVolumeStatus(client, id, nodeID, nodePool, opts); err != nil {
+			c.Ui.Error(err.Error())
+			return 1
+		}
+	case "":
+		if id == "" {
+			// for list, we want to show both
+			dhvErr := c.hostVolumeList(client, nodeID, nodePool, opts)
+			if dhvErr != nil {
+				c.Ui.Error(dhvErr.Error())
+			}
+			c.Ui.Output("")
+			csiErr := c.csiVolumesList(client, opts)
+			if csiErr != nil {
+				c.Ui.Error(csiErr.Error())
+			}
+			if dhvErr == nil && csiErr == nil {
+				return 0
+			}
+			return 1
+		} else {
+			// for read, we only want to show whichever has results
+			hostErr := c.hostVolumeStatus(client, id, nodeID, nodePool, opts)
+			if hostErr != nil {
```

Comment:
not a big deal, but this assumes that _any_ error equates to "not found"

Reply:
That's a good point. Once we've got at least one prefix match we know there's no reason to continue on to CSI. Let me see what I can to do tighten that up.

Reply:
Fixed by returning a wrapped error we can check in the caller and bail out early. Example:

```
$ nomad volume status 3
Prefix matched multiple host volumes

ID        Name              Namespace  Plugin ID  Node ID   Node Pool  State
3a8bcb6b  internal-plugin2  default    mkdir      11bfea4e  default    ready
3ddbaaac  internal-plugin7  default    mkdir      11bfea4e  default    ready
```